### ISSUE TITLE
newer version ollama chat more slower

### ISSUE THREAD

**Issue by krmao** [ollama/ollama#7807]
### What is the issue?

with the same code on the same machine
Apple M2 Pro
macos 15.1.1 (24B91)

```python
import time
import ollama

start_time = time.perf_counter()
#len(final_chat_messages)= 6107
ai_response = ollama.chat(model=model, messages=final_chat_messages, tools=TOOLS)
print(f'time after ollama.chat: {((time.perf_counter() - start_time) * 1000):.0f}ms')
```
after a lots of test:

0.3.14 only need 1s,  first time need 4580ms, but then only need 1073ms with the same question

0.4.2 need 20s+, 

0.4.3-0.4.4  need 10s

### OS

macOS

### GPU

Apple

### CPU

Apple

### Ollama version

0.4.2

**Comment by jmorganca** [comment]
@krmao thanks for the issue. May I ask what the prompt was (or how long it was?) This will help debug the performance issues.

**Comment by krmao** [comment]
can be found at the code `#len(final_chat_messages)= 6107`, about 6107 chars,    the mode is `dwightfoster03/functionary-small-v3.1:latest`

**Comment by krmao** [comment]
I have a new found,  `0.3.14 ` first time need 4580ms, but then only need 1073ms with the same question

if change to the different question, then need 4493ms,  seems not stable. I have not attention this on weeks ago. 
it works everything ok before I am upgrading to the latest version.

**Comment by rick-github** [comment]
Have you adjusted the size of the context window (`num_ctx`)?  0.4.3+ handle long messages differently to 0.3.* when they exceed context.

**Comment by krmao** [comment]
> Have you adjusted the size of the context window (`num_ctx`)? 0.4.3+ handle long messages differently to 0.3.* when they exceed context.

yes, I have the code to add history question and response

**Comment by rick-github** [comment]
If you could provide a complete script to demonstrate the problem you are seeing it would make debugging much easier.

**Comment by krmao** [comment]
In the process of creating a test code, I discovered that the system prompt might have been added twice, resulting in duplication. After removing one of them, each inference took less than 1 second, even for different questions.

This revealed the root cause of the issue: the doubled prompt string exceeded 10,000 characters. This highlights the differences in handling very long prompts between version 0.3.x and 0.4.x.

Apologies for taking up everyone's time. The issue has now been resolved. Thank you all!
